app/crud.py

```
@@ -0,0 +1,59 @@
+        raise HTTPException(status_code=500, detail=f"Error retrieving ToDos: {e}")
+
+
+async def get_todo(db: AsyncSession, todo_id: int):
+    try:
+        result = await db.execute(select(ToDo).where(ToDo.id == todo_id))
+        todo = result.scalar_one()
+        return todo
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Error retrieving ToDo with ID:{todo_id}: {e}"
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)